### ISSUE TITLE
Error creating overlay mount for JDK 16 images

### ISSUE THREAD

**Issue by neelkamath** [keeganwitt/docker-gradle#189]
Here's an example of JDK 11 working:

```
$ docker run --rm -it gradle:7.0.2-jdk11-hotspot
```

Here's an example of JDK 16 (not) working:

```
$ docker run --rm -it gradle:7.0.2-jdk16-hotspot
docker: Error response from daemon: error creating overlay mount to /var/lib/docker/overlay2/a7990296c2dfdcd57cf137405cbb4b464cd0211cc60c79b303565308bb74beb2-init/merged: invalid argument.
See 'docker run --help'.
```

Info:
- Computer: MacBook
- OS: macOS 11.2.3
- Chip: Intel
- Docker engine: v20.10.6

**Comment by keeganwitt** [comment]
Is this resolved now?

I was able to run `docker run --rm gradle:7.0.2-jdk16-hotspot gradle --no-daemon --version` on both Windows 10 and MacOS 11.3.1 (Docker version 20.10.6 build 379c289 and Intel chip for both).

**Comment by neelkamath** [comment]
No, running `docker run --rm gradle:7.0.2-jdk16-hotspot gradle --no-daemon --version` doesn't work either. The only difference I see is that you're using macOS 11.3 instead of macOS 11.2. It's a long story but there's no practical way for me to update my macOS. I guess if it works for you, then this isn't a Gradle issue but I don't know what else it could be then because the other Docker images (including many Gradle ones) work for me.

**Comment by keeganwitt** [comment]
Oh, I didn't mention it, but this was Docker Desktop and not Docker Toolbox (though I'm guessing that was the case for you as well).  Having trouble thinking what else might be different.

You did try pulling the image before trying again, right?

**Comment by neelkamath** [comment]
Yes, I'm using Docker Desktop, and not Docker Toolbox.
Yes, I pulled the image again. The `docker pull` command works but the error occurs when I use the `docker run` command. I even tried reinstalling Docker but that didn't help either.

The only reason I can think of is because I'm using macOS 11.2 instead of a newer version. I've tried for a few days now but I simply cannot update macOS without reinstalling the entire system, and I'm tired of reinstalling the OS every time there's an update.

Since I can manage to live without JDK 16 for now, I'll continue using an older image, and hope that it'll work once I eventually get a new laptop.
I'll try to remember to follow up here if I ever find a resolution though in case someone else is facing the same problem. I'll leave it up to you whether you want to close this issue.
